src/camvidlog2/yoloe/ai.py

```
@@ -0,0 +1,214 @@
+    confidences = np.max(class_scores, axis=1)
+    # Stack boxes, confidences, and class_ids: shape (N, 6)
+    results_top = np.hstack([boxes, confidences[:, None], class_ids[:, None]])
+    # Filter by confidence threshold
+    results_filtered_conf = results_top[results_top[:, 4] > conf]
+    # Convert (x_center, y_center, width, height) to (x1, y1, x2, y2) in-place
+    convert_xyhw_into_x1y1x2y2(results_filtered_conf)
+    # Scale coordinates from 640x640 to original image size in-place
+    apply_scaling(results_filtered_conf, img_w, img_h)
+    # Apply Non-Maximum Suppression (NMS)
+    results_filtered_conf_nms = nms(results_filtered_conf, nms_thresh)
+    return results_filtered_conf_nms
```

Ruff (0.11.9)

Unnecessary assignment to `results_filtered_conf_nms` before `return` statement

Remove unnecessary assignment

(RET504)